packages/datadog-instrumentations/src/jest.js

```diff
@@ -1267,6 +1271,24 @@ addHook({
 }, (runtimePackage) => {
   const Runtime = runtimePackage.default ?? runtimePackage
 
+  shimmer.wrap(Runtime.prototype, '_createJestObjectFor', _createJestObjectFor => function (from) {
+    const result = _createJestObjectFor.apply(this, arguments)
+    const suiteFilePath = this._testPath
+
+    shimmer.wrap(result, 'mock', mock => function (moduleName, mockFactory, options) {
+      if (suiteFilePath) {
+        const testSuite = getTestSuitePath(suiteFilePath, process.cwd())
+        const existingMockedFiles = testSuiteMockedFiles.get(testSuite) || []
+        const suiteDir = path.dirname(suiteFilePath)
+        const mockPath = path.resolve(suiteDir, moduleName)
+        const relativeMockPath = path.relative(process.cwd(), mockPath)
+        testSuiteMockedFiles.set(testSuite, [...existingMockedFiles, relativeMockPath])
+      }
+      return mock.apply(this, arguments)
```

Comment:
`suiteFilePath` is the absolute test suite path, right? I wonder if we could simplify things by keeping the absolute path everywhere and only calculating the relative path before `testSuiteCodeCoverageCh.publish`
```suggestion
      if (suiteFilePath) {
        const existingMockedFiles = testSuiteMockedFiles.get(suiteFilePath) || []
        const suiteDir = path.dirname(suiteFilePath)
        const mockPath = path.resolve(suiteDir, moduleName)
        existingMockedFiles.push(mockPath)
      }
      return mock.apply(this, arguments)
```
caveat: `testSuiteMockedFiles` now includes absolute paths, so when extracting `mockedFiles` you should use the absolute path

```javascript
const mockedFiles = getFilesWithPath(testSuiteMockedFiles.get(environment.testSuiteAbsolutePath) || [])
```

`environment.testSuiteAbsolutePath` would be undefined but I think you can add it simply by doing something like 

```javascript
this.testSuiteAbsolutePath = context.testPath
```

in the constructor.

Reply:
Already refactored :)